dev/com.ibm.ws.cdi.visibility_fat/fat/src/com/ibm/ws/cdi/visibility/tests/vistest/VisTestAddWarCPToEar.java

```diff
@@ -0,0 +1,330 @@
+import static com.ibm.ws.cdi.visibility.tests.vistest.VisTestSetup.doTestVisibilityFromCommonLib;
+import static com.ibm.ws.cdi.visibility.tests.vistest.VisTestSetup.doTestVisibilityFromEarLib;
+import static com.ibm.ws.cdi.visibility.tests.vistest.VisTestSetup.doTestVisibilityFromEjb;
+import static com.ibm.ws.cdi.visibility.tests.vistest.VisTestSetup.doTestVisibilityFromEjbAppClientLib;
+import static com.ibm.ws.cdi.visibility.tests.vistest.VisTestSetup.doTestVisibilityFromEjbAsAppClientLib;
+import static com.ibm.ws.cdi.visibility.tests.vistest.VisTestSetup.doTestVisibilityFromEjbAsEjbLib;
+import static com.ibm.ws.cdi.visibility.tests.vistest.VisTestSetup.doTestVisibilityFromEjbAsWarLib;
+import static com.ibm.ws.cdi.visibility.tests.vistest.VisTestSetup.doTestVisibilityFromEjbLib;
+import static com.ibm.ws.cdi.visibility.tests.vistest.VisTestSetup.doTestVisibilityFromEjbWarLib;
+import static com.ibm.ws.cdi.visibility.tests.vistest.VisTestSetup.doTestVisibilityFromPrivateLib;
+import static com.ibm.ws.cdi.visibility.tests.vistest.VisTestSetup.doTestVisibilityFromRuntimeExtRegular;
+import static com.ibm.ws.cdi.visibility.tests.vistest.VisTestSetup.doTestVisibilityFromRuntimeExtSeeAll;
+import static com.ibm.ws.cdi.visibility.tests.vistest.VisTestSetup.doTestVisibilityFromStandaloneWar;
+import static com.ibm.ws.cdi.visibility.tests.vistest.VisTestSetup.doTestVisibilityFromStandaloneWarLib;
+import static com.ibm.ws.cdi.visibility.tests.vistest.VisTestSetup.doTestVisibilityFromWar;
+import static com.ibm.ws.cdi.visibility.tests.vistest.VisTestSetup.doTestVisibilityFromWarAppClientLib;
+import static com.ibm.ws.cdi.visibility.tests.vistest.VisTestSetup.doTestVisibilityFromWarLib;
+import static com.ibm.ws.cdi.visibility.tests.vistest.VisTestSetup.doTestVisibilityFromWarWebinfLib;
+
+import java.util.Map;
+import java.util.logging.Logger;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.ws.cdi.visibility.tests.vistest.VisTestSetup.Location;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.custom.junit.runner.Mode;
+import componenttest.custom.junit.runner.Mode.TestMode;
+import componenttest.topology.impl.LibertyClient;
+import componenttest.topology.impl.LibertyClientFactory;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.utils.FATServletClient;
+
+@Mode(TestMode.FULL)
+@RunWith(FATRunner.class)
+public class VisTestAddWarCPToEar extends FATServletClient {
+
+    public static final String SERVER_NAME = "visTestServerAddWarCPToEar";
+    public static final String CLIENT_NAME = "visTestClientAddWarCPToEar";
+
+//    @ClassRule
+//    public static RepeatTests r = EERepeatActions.repeat(SERVER_NAME, EERepeatActions.EE10, EERepeatActions.EE11, EERepeatActions.EE8, EERepeatActions.EE7);
```

Comment:
I don't think so because the behavior here isn't specific to any EE version.  I should remove that commented out rule unless you see value in repeating it.

Reply:
No, I think you can remove it